exchanges/orderbook/incremental_updates.go

```diff
@@ -0,0 +1,234 @@
+package orderbook
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+// Action defines a set of differing states required to implement an incoming orderbook update used in conjunction with
+// UpdateEntriesByID
+type Action uint8
+
+const (
+	// Amend applies amount adjustment by ID
+	Amend Action = iota + 1
+	// Delete removes price level from book by ID
+	Delete
+	// Insert adds price level to book
+	Insert
+	// UpdateInsert on conflict applies amount adjustment or appends new amount to book
+	UpdateInsert
+)
```

Comment:
🚧 
* Comment can be cleaner, since how we identify records which have IDs seems obvious
* Comments on individual consts seem redundant too
* Amend vs Update consistency; Update more consistent with CRUD; Admitting that Insert is better here
* I often regret not suffixing types like this, and their consts, because they're sitting on method/func names
```suggestion
// ActionType defines the behaviour of an orderbook update
type ActionType uint8

// ActionType constants for use with ProcessUpdate
const (
	UnknownAction ActionType = iota
	InsertAction
	UpdateOrInsertAction
	UpdateAction
	DeleteAction
)
```